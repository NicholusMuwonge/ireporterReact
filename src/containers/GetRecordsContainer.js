/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import RecordFetchComponent from '../components/records/Recordfetch';
import { getRecords } from '../actions/GetRecordsActions';
import Authenticated from '../routes/Authenticated';


export class getRecordsContainer extends Component {
  componentDidMount() {
    const { recordAction } = this.props;
    recordAction();
  }

  render() {
    const { record, loading } = this.props;
    if (loading) {
      return (
        <div
          className="spinner-border"
          role="status"
          style={{
            fontSize: '130px', marginTop: '20%', marginLeft: '50%', color: 'teal',
          }}
        >
          <span className="sr-only">Loading...</span>
        </div>
      );
    }

    return (
      <RecordFetchComponent record={record} loading={loading} />
    );
  }
}

getRecordsContainer.propTypes = {
  recordAction: PropTypes.func.isRequired,
};
export const mapStateToProps = state => ({
  record: state.recordsReducer.record,
  loading: state.recordsReducer.loading,
});

export default connect(
  mapStateToProps, { recordAction: getRecords },
)(Authenticated(getRecordsContainer));
