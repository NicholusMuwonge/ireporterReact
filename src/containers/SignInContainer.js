/* eslint-disable react/no-unused-state */
/* eslint-disable no-unused-vars */
/* eslint-disable no-console */
/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Login from '../components/Login/Login';
import { loginAction } from '../actions/SigninActions';
import { AlreadyAuthenticated } from '../routes/Authenticated';


export class SignInContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      userName: '',
      email: '',
      password: '',
      buttonLoading: false,
    };
    this.submitHandler = this.submitHandler.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  fetchData = () => {
    this.setState({ buttonLoading: true });

    setTimeout(() => {
      this.setState({ buttonLoading: false });
    }, 5000);
  };

  handleChange(e) {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value });
  }

  submitHandler(e) {
    e.preventDefault();
    this.fetchData();
    const {
      password, userName,
    } = this.state;

    const { loginAction } = this.props;
    const data = {
      user_name: userName,
      user_password: password,
    };
    loginAction(this.props, data);
  }

  render() {
    const {
      password, userName, loading, buttonLoading,
    } = this.state;
    return (

      <Login
        submitHandler={this.submitHandler}
        handleChange={this.handleChange}
        password={password}
        userName={userName}
        action={loginAction}
        isLoading={loading}
        buttonLoading={buttonLoading}
      />

    );
  }
}

export const mapStateToProps = state => ({
  loading: state.SignInReducers.loading,
});

export default connect(
  mapStateToProps,
  { loginAction },
)(AlreadyAuthenticated(SignInContainer));
