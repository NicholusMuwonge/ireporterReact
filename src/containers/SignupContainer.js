/* eslint-disable no-unused-vars */
/* eslint-disable no-console */
/* eslint-disable react/prop-types */
/* eslint-disable no-shadow */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import SignUp from '../components/Signup/Signup';
import { SignUpAction } from '../actions/SignupAction';
import { AlreadyAuthenticated } from '../routes/Authenticated';


export class SignUpContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      userName: '',
      email: '',
      password: '',
      password2: '',
      buttonLoading: false,
    };
    this.submitHandler = this.submitHandler.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  fetchData = () => {
    this.setState({ buttonLoading: true });

    setTimeout(() => {
      this.setState({ buttonLoading: false });
    }, 5000);
  };

  handleChange(e) {
    e.preventDefault();
    this.setState({ [e.target.name]: e.target.value });
  }

  submitHandler(e) {
    e.preventDefault();
    this.fetchData();
    const {
      password, password2, userName, email,
    } = this.state;
    const { SignUpAction } = this.props;
    if (password === password2) {
      const data = {
        user_name: userName,
        email,
        user_password: password,
      };
      SignUpAction(this.props, data);
    } else {
      toast.dismiss();
      toast.error('Passwords do not match!', {
        position: toast.POSITION.CENTER_RIGHT,
        autoClose: 3000,
      });
    }
  }

  render() {
    const {
      password, password2, userName, email, loading, buttonLoading,
    } = this.state;
    return (

      <SignUp
        submitHandler={this.submitHandler}
        handleChange={this.handleChange}
        password={password}
        password2={password2}
        email={email}
        userName={userName}
        action={SignUpAction}
        isLoading={loading}
        buttonLoading={buttonLoading}
      />

    );
  }
}

export const mapStateToProps = state => ({
  loading: state.SignUpReducers.loading,
});

export default connect(
  mapStateToProps,
  { SignUpAction },
)(AlreadyAuthenticated(SignUpContainer));
