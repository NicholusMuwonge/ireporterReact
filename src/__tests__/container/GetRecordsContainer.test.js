/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import { getRecordsContainer, mapStateToProps } from '../../containers/GetRecordsContainer';

describe.only('<getRecordsContainer />', () => {
  let wrapper;
  const props = {
    record: { },
    loading: false,
    recordAction: jest.fn(),
  };
  beforeEach(() => {
    wrapper = shallow(<getRecordsContainer {...props} />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should map the comments reducer to props', () => {
    const state = {
      recordsReducer: {
        record: {},
        loading: false,
      },
    };
    expect(mapStateToProps(state)).toEqual({
      record: {},
      loading: false,

    });
  });
});
