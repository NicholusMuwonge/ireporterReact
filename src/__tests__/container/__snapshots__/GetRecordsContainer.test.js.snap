// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<getRecordsContainer /> should match the snapshot 1`] = `ShallowWrapper {}`;
